app/registries/foreman/plugin/report_scanner_registry.rb

```diff
@@ -0,0 +1,38 @@
+require_dependency File.expand_path('../../../../services/report_scanner/puppet_report_scanner', __FILE__)
+
+module Foreman
+  class Plugin
+    class ReportScannerRegistry
+      DEFAULT_REPORT_SCANNER = [
+        ::Foreman::PuppetReportScanner
+      ].freeze
+
+      attr_accessor :report_scanner
+
+      def initialize
+        @report_scanner = []
+        register_default_scanner
+      end
+
+      def report_scanner
```

Comment:
if this method returns multiple scanners, would it be better to pluralize the name?